src/file.c

```diff
@@ -157,7 +157,9 @@ FILE *openWritableFile(const char *pathname, const char *mode)
 #else // the rest of the platforms just have different base paths
 
 #if defined(_3DS)
-#define FILE_BASE_PATH "sdmc:/OpenSupaplex/"
+#include <sys/stat.h>
+#define FILE_BASE_PATH "romfs:/"
+#define FILE_BASE_WRITABLE_PATH "sdmc:/3ds/OpenSupaplex/"
```

Comment:
I think I prefer to keep this "generic" in the root of the SD card: it'd be weird to have it installed as a CIA and have to look for the data inside of the `3ds` folder…
```suggestion
#define FILE_BASE_WRITABLE_PATH "sdmc:/OpenSupaplex/"
```

Reply:
The ```3ds``` folder would be the default folder for homebrew on the 3ds.
While this is in no means a requirement, it's being used for almost* every homebrew developed for 3ds.
Opening the homebrew launcher will also default to the ```3ds``` folder, it would be nice if this is listed instead of having to navigate back to root.

This definitely has my preference, if you wish to have it reverted i'll be more than happy to oblige.

Reply:
That was definitely not what I saw when I was looking for examples of "common paths for data". However, I might have picked the wrong sample of homebrew games 😅 

In any case, I think there is a very good reason to keep in the root folder: users of older versions of OpenSupaplex would (apparently) lose their saved games unless they move their existing `/OpenSupaplex` folder into `/3ds` 😕 

So yes, please, let's keep it at the root of the SD card. I'm sorry if this was a poor choice to begin with 😅

Reply:
The current used resource files on the sd will become redundant with this change. They should be removed, without removing the save files. If the user should be guided how to do that, we may as well guide them to move the saves.

As in: 
- Move ```ADVANCED.CFG```, ```HALLFAME.LST``` and ```PLAYER.LST```
- Delete the rest

instead of:
- Backup ```ADVANCED.CFG```, ```HALLFAME.LST``` and ```PLAYER.LST```
- Delete the rest


Are you sure?

Reply:
Good point!! To be honest, I'm not very worried about that. Existing users are free to do whatever they want with the existing (soon unused) files, as long as my releases don't lose anything they had 😅 

In any case, I can mention this in the release notes, to give them a chance to clean it up if they want 😄

Reply:
Should i mark this as resolved? Or is any further action required?

There's the possibility to check if we're running a .3dsx and use a relative path to the executable.
The .cia build will always default to root.

Necessary? Not really, in my opinion.

Reply:
In order to keep backwards compatibility, let's apply the change I suggested:
```suggestion
#define FILE_BASE_WRITABLE_PATH "sdmc:/OpenSupaplex/"
```

And therefore, the suggestion in `3ds/ci-build-3ds.sh` must be applied too.